mantidproject/mantid
# Update cppcheck to 2.16.0

### Description of work
Update to cppcheck 2.16.0.
- It has been restricted to be installed only in Unix  (MacOS and Linux) `mantid-developer` environments since 2.16.0 is not currently available for Windows.
- `class_X_Y` warnings have been silenced because they seemed to be meaningless.
- The tool to generate the suppressions now sorts by file name and line number.
- Cppcheck internal errors are separated from the list of ones that we can fix.

Fixes #37906 

### To test:
PR should pass checks. Ensure that cppcheck was run. I made a trivial change to a `.cpp` file to force it to run.

The list of defects can be seen here:
https://builds.mantidproject.org/job/pull_requests-cppcheck/62813/CppCheck/
Check that they look reasonable.

*This does not require release notes* because **it's an update to a code quality checking tool**




---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
